server/database_test.go

```diff
@@ -167,6 +178,50 @@ func TestGetApplicationDeployments(t *testing.T) {
 	}
 }
 
+func TestGetApplicationDeploymentsByTarget(t *testing.T) {
+	db := newTestDb(t)
+	defer cleanCloseTestDb(db, t)
+
+	firstDeployment := buildDeployment(9999)
+	err := createDeployment(db, firstDeployment)
+	checkErr(t, err)
+
+	secondDeployment := buildDeployment(9999)
+	err = createDeployment(db, secondDeployment)
+	checkErr(t, err)
+
+	thirdDeployment := buildDeploymentOtherTarget(9999)
+	err = createDeployment(db, thirdDeployment)
+	checkErr(t, err)
+
+	application := &models.Application{Name: "flincOnRails"}
+
+	deployments, err := getApplicationDeployments(db, application, 99)
+	checkErr(t, err)
+
+	if len(deployments) != 3 {
+		t.Errorf("Wrong number of deployments returned. expected=%d, got=%d", 3, len(deployments))
+	}
+
+	deployments, err = getApplicationDeploymentsByTarget(db, application, "production")
+	checkErr(t, err)
+	if len(deployments) != 2 {
+		t.Errorf("Wrong number of deployments returned. expected=%d, got=%d", 2, len(deployments))
+	}
+
+	deployments, err = getApplicationDeploymentsByTarget(db, application, "test")
+	checkErr(t, err)
+	if len(deployments) != 1 {
+		t.Errorf("Wrong number of deployments returned. expected=%d, got=%d", 1, len(deployments))
+	}
+
+	deployments, err = getApplicationDeploymentsByTarget(db, application, "empty")
+	checkErr(t, err)
+	if len(deployments) != 0 {
+		t.Errorf("Wrong number of deployments returned. expected=%d, got=%d", 0, len(deployments))
+	}
+}
```

Comment:
Nice! I like this test!

One thought though: maybe we can get rid of the `buildDeploymentOtherTarget` function if we just use `buildDeployment` and then modify the deployment in place like so:

```go
deployment := buildDeployment(9999)
deployment.TargetName = "foobar"
err = createDeployment(db, deployment)
```

I think that would make the test more easy to understand since all the relevant information is in those lines.

What do you think?